source/HailPanel.cpp

```diff
@@ -55,25 +55,18 @@ HailPanel::HailPanel(PlayerInfo &player, const shared_ptr<Ship> &ship)
 		message = "(There is no response to your hail.)";
 	else if(!hasLanguage)
 		message = "(An alien voice says something in a language you do not recognize.)";
-	else if(gov->IsEnemy())
+	else if(gov->IsEnemy() && !ship->IsDisabled())
```

Comment:
@Elyssaen most likely just wanted to fuse the two nested conditions `if(gov->IsEnemy())` and `if(!ship->IsDisabled())` into one after removing the `if(ship->IsDisabled())` branch inside the IsEnemy() branch.
The `IsDisabled()` branch gets removed because the new intention is that this function does not return any result in that case, and after removing it, only the `else` branch remains and thus it seemed logic to merge both nested if()s

That's also the reason why the next `else if()` had to be extended by `!gov->isEnemy()`, to prevent those ships being handled there.
In terms of a nice, readable, and understandable code, I agree with @tehhowch that a proper comment would be nice. Ideally, that comment just could replace the original `message = GameData:: ...` statement.
But then, we should also do the same with the `else message = "Our ship has been disabled! ..."` lines I removed for my friendlyDisabledHail part.